ndb/src/google/cloud/ndb/model.py

```diff
@@ -2144,9 +2150,154 @@ def __init__(self, *args, **kwargs):
+        property values are updated to match what was written. Finally, beware
+        that this also updates the value in the in-process cache, **and** that
+        ``auto_now_add`` may interact weirdly with transaction retries (a retry
+        of a property with ``auto_now_add`` set will reuse the value that was
+        set on the first try).
+
+    .. automethod:: _validate
+    .. automethod:: _prepare_for_put
+
+    Args:
+        name (str): The name of the property.
+        auto_now (bool): Indicates that the property should be set to the
+            current datetime when an entity is created and whenever it is
+            updated.
+        auto_now_add (bool): Indicates that the property should be set to the
+            current datetime when an entity is created.
+        indexed (bool): Indicates if the value should be indexed.
+        repeated (bool): Indicates if this property is repeated, i.e. contains
+            multiple values.
+        required (bool): Indicates if this property is required on the given
+            model type.
+        default (bytes): The default value for this property.
+        choices (Iterable[bytes]): A container of allowed values for this
+            property.
+        validator (Callable[[~google.cloud.ndb.model.Property, Any], bool]): A
+            validator to be used to check values.
+        verbose_name (str): A longer, user-friendly name for this property.
+        write_empty_list (bool): Indicates if an empty list should be written
+            to the datastore.
+
+    Raises:
+        ValueError: If ``repeated=True`` and ``auto_now=True``.
+        ValueError: If ``repeated=True`` and ``auto_now_add=True``.
+    """
+
+    _auto_now = False
+    _auto_now_add = False
+
+    def __init__(
+        self,
+        name=None,
+        *,
+        auto_now=None,
+        auto_now_add=None,
+        indexed=None,
+        repeated=None,
+        required=None,
+        default=None,
+        choices=None,
+        validator=None,
+        verbose_name=None,
+        write_empty_list=None
+    ):
+        super(DateTimeProperty, self).__init__(
+            name=name,
+            indexed=indexed,
+            repeated=repeated,
+            required=required,
+            default=default,
+            choices=choices,
+            validator=validator,
+            verbose_name=verbose_name,
+            write_empty_list=write_empty_list,
+        )
+        if self._repeated:
+            if auto_now:
+                raise ValueError(
+                    "DateTimeProperty {} could use auto_now and be "
+                    "repeated, but there would be no point.".format(self._name)
+                )
+            elif auto_now_add:
+                raise ValueError(
+                    "DateTimeProperty {} could use auto_now_add and be "
+                    "repeated, but there would be no point.".format(self._name)
+                )
+        if auto_now is not None:
```

Comment:
This change differs from the original (i.e. checking before setting)

Reply:
:+1: